### PR TITLE
Implement unique+array parts of 24024

### PR DESCRIPTION
**Pull request by jbrockmendel**
They have tests that go with them and are reasonably self-contained.

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -1556,9 +1555,17 @@ def unique(self):
...
-            data type, returned as a Categorical.
+        ndarray or ExtensionArray
+            The unique values returned as a NumPy array. In case of an
+            extension-array backed Series, a new
+            :class:`~api.extensions.ExtensionArray` of that type with just
+            the unique values is returned. This includes
+
+            * Categorical
+            * Period
+            * Datetime with Timezone
```
Comment (jreback) [inline pandas/core/series.py]: IntegerNA ?

**Review by jreback — COMMENTED** [review]
small comment

File: pandas/core/series.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/series.py]: will edit.  note otherwise this is verbatim from 24024

File: pandas/tests/test_base.py [hunk 2]
```diff
@@ -383,8 +387,12 @@ def test_value_counts_unique_nunique(self):
                 assert result[0] == orig[0]
                 for r in result:
                     assert isinstance(r, Timestamp)
-                tm.assert_numpy_array_equal(result,
-                                            orig._values.astype(object).values)
+
+                # TODO(#24024) once orig._values returns DTA, remove
+                #  the `._eadata` below
```
Comment (jreback) [inline pandas/tests/test_base.py]: in future, can make this a proper form test (e.g. result =, expected = )

File: pandas/tests/test_base.py [hunk 3]
```diff
@@ -431,7 +441,7 @@ def test_value_counts_unique_nunique_null(self):
                     o = klass(values.repeat(range(1, len(o) + 1)))
                     o.name = 'a'
                 else:
-                    if is_datetime64tz_dtype(o):
+                    if isinstance(o, DatetimeIndex):
                         expected_index = orig._values._shallow_copy(values)
```
Comment (jreback) [inline pandas/tests/test_base.py]: these are super-hacks, in future, let's try to remove

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**